pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
```

Comment:
They should not be hardcoded, rather than using whatever the `ISTIO_MUTUAL/STRICT` mode today's configuration, including all the root, san verification, details. We need to plumb the config we have from `applyUpstreamTlsSettings`.